app/views/ProjectEdit/GeoAreas/RegionsMap/index.tsx

```diff
@@ -63,44 +74,62 @@ function RegionsMap(props: Props) {
     const [regionOptions, setRegionOptions] = useState<Region[] | null | undefined>(undefined);
     const alert = useAlert();
 
-    interface RegionPatchCtx {
-        newRegion: number,
-        body: {
-            regions: { id: number | string }[],
+    const [
+        patchRegion,
+        {
+            loading: pendingPatchRegion,
         },
-    }
-
-    const {
-        trigger: regionPatchTrigger,
-    } = useLazyRequest<ProjectDetails, RegionPatchCtx>({
-        url: `server://projects/${projectId}/`,
-        method: 'PATCH',
-        body: (ctx) => ctx.body,
-        onSuccess: (_, ctx) => {
-            alert.show(
-                'Successfully added geo area to the project.',
-                { variant: 'success' },
-            );
-            onActiveRegionChange(ctx.newRegion.toString());
-            onRegionAdd();
+    ] = useMutation<PatchRegionMutation, PatchRegionMutationVariables>(
+        PATCH_REGION,
+        {
+            onCompleted: (response) => {
+                if (!response.project?.projectRegionBulk) {
+                    return;
+                }
+
+                const {
+                    result,
+                    errors,
+                } = response.project.projectRegionBulk;
+
+                if (errors) {
+                    alert.show(
+                        'Failed to publish selected region.',
+                        { variant: 'error' },
+                    );
+                }
+
+                const ok = isDefined(result) && result?.length > 0;
+
+                if (ok) {
+                    alert.show(
+                        'Region is successfully added!',
+                        { variant: 'success' },
+                    );
+                    onRegionAdd();
+                }
+            },
+            onError: () => {
+                alert.show(
+                    'Failed to add selected region.',
+                    { variant: 'error' },
+                );
+            },
         },
-    });
+    );
 
     const handleAddRegionConfirm = useCallback(
         (value: number | undefined) => {
-            if (regions && value) {
-                regionPatchTrigger({
-                    newRegion: value,
-                    body: {
-                        regions: [
-                            ...regions.map((d) => ({ id: d.id })),
-                            { id: value },
-                        ],
+            if (value) {
+                patchRegion({
+                    variables: {
+                        projectId,
+                        regionId: [String(value)],
```

Comment:
Won't this replace the previous regions and only set one?

Reply:
Mutation only requires selected region for the patch not the entire list.

Reply:
Okayy, thenks